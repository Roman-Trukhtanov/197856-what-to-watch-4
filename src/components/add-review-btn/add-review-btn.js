import React from "react";
import PropTypes from "prop-types";
import {ActionCreator as ScreenActionCreator} from "../../reducer/screen/screen";
import {ScreenType} from "../../const";
import {connect} from "react-redux";

const AddReviewBtn = (props) => {
  const {
    onAddReviewBtnClick,
  } = props;

  return (
    <a href="/" className="btn movie-card__button" onClick={(evt) => {
      evt.preventDefault();
      onAddReviewBtnClick();
    }}>Add review</a>
  );
};

AddReviewBtn.propTypes = {
  onAddReviewBtnClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onAddReviewBtnClick() {
    dispatch(ScreenActionCreator.changeScreen(ScreenType.ADD_REVIEW));
  }
});

export {AddReviewBtn};

export default connect(null, mapDispatchToProps)(AddReviewBtn);
