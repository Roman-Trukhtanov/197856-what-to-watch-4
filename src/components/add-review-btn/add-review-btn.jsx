import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {ScreenType} from "../../const";

const AddReviewBtn = (props) => {
  const {movie} = props;

  return (
    <Link to={`/${ScreenType.FILMS}/${movie.id}/${ScreenType.REVIEW}`} className="btn movie-card__button">Add review</Link>
  );
};

AddReviewBtn.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number.isRequired,
  }).isRequired,
};

export {AddReviewBtn};

export default AddReviewBtn;
