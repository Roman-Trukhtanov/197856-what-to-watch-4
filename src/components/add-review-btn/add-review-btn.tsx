import React from "react";
import {Link} from "react-router-dom";
import {ScreenType} from "../../const";
import {Movie} from "../../types";

interface Props {
  movie: Movie;
}

const AddReviewBtn: React.FunctionComponent<Props> = (props: Props) => {
  const {movie} = props;

  return (
    <Link to={`/${ScreenType.FILMS}/${movie.id}/${ScreenType.REVIEW}`} className="btn movie-card__button">Add review</Link>
  );
};

export {AddReviewBtn};

export default AddReviewBtn;
