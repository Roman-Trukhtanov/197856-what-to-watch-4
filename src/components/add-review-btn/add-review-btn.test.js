import React from "react";
import renderer from "react-test-renderer";
import {AddReviewBtn} from "./add-review-btn";

describe(`AddReviewBtn component`, () => {
  it(`Render AddReviewBtn`, () => {
    const tree = renderer
      .create(<AddReviewBtn
        onAddReviewBtnClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
