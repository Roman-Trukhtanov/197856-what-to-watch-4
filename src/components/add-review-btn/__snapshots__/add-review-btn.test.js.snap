// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReviewBtn component Render AddReviewBtn 1`] = `
<a
  className="btn movie-card__button"
  href="/"
  onClick={[Function]}
>
  Add review
</a>
`;
